src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2153,6 +2153,7 @@ NetworkOPsImp::pubValidation(std::shared_ptr<STValidation> const& val)
         jvObj[jss::flags] = val->getFlags();
         jvObj[jss::signing_time] = *(*val)[~sfSigningTime];
         jvObj[jss::data] = strHex(val->getSerializer().slice());
+        jvObj[jss::network_id] = app_.config().NETWORK_ID;
```

Comment:
Does this data need to be added to the `val` object?

Reply:
`network_id` is returned in the `server_info` command output. It is computed in a slightly different way at https://github.com/XRPLF/rippled/blob/350d213ee8a709e0ed417da0f081291ed411480a/src/ripple/app/misc/NetworkOPs.cpp#L2523

I understand that this information is very convinient in the Validation stream, but can't we grab it from the `server_info` command too?

Reply:
Making a call to server info is not always possible because not all validators expose their IPs.

Reply:
Most don't (especially UNL validators) for security reasons.

Reply:
okay, I didn't realise that, thanks!